### PR TITLE
[TUTORIAL] Improve gemm tutorial

### PR DESCRIPTION
**Pull request by yuruofeifei**
Added more optimization with help from @yidawang , current matrix mutiplication speed is about 60% of MKL.

### CODE REVIEW

File: tutorials/optimize/opt_gemm.py [hunk 1]
```diff
@@ -42,45 +43,51 @@
 import numpy
 import timeit
 
-# The size of the square matrix
-N = 1024
+# The size of the matrix
+N1 = 1024
+N2 = 1024
```
Comment (yidawang) [inline tutorials/optimize/opt_gemm.py]: I am thinking about making it more generic by having `M, N, K` here, so we have `[M, K] * [K, N] = [M, N]`. And we should make a note here saying that we are testing out on the square matrices only; the users should feel free to change the matrix shapes; and different shapes yield different results, sometimes TVM outperforms MKL.

File: tutorials/optimize/opt_gemm.py [hunk 2]
```diff
@@ -42,45 +43,51 @@
...
 # The default tensor type in tvm
 dtype = "float32"
+
+target = 'llvm -mcpu=core-avx2'
```
Comment (yidawang) [inline tutorials/optimize/opt_gemm.py]: Leave a comment of using this advanced ISA.

File: tutorials/optimize/opt_gemm.py [hunk 3]
```diff
@@ -42,45 +43,51 @@
...
+ctx = tvm.context(target, 0)
+
 # Random generated tensor for testing
-a = tvm.nd.array(numpy.random.rand(N, N).astype(dtype), tvm.cpu(0))
-b = tvm.nd.array(numpy.random.rand(N, N).astype(dtype), tvm.cpu(0))
+a = tvm.nd.array(numpy.random.rand(N1, N2).astype(dtype), ctx)
+b = tvm.nd.array(numpy.random.rand(N2, N1).astype(dtype), ctx)
```
Comment (yidawang) [inline tutorials/optimize/opt_gemm.py]: Then this should be `M, K` and `K, N`

Comment (yidawang) [reply tutorials/optimize/opt_gemm.py]: Similar things apply to the whole script.

File: tutorials/optimize/opt_gemm.py [hunk 4]
```diff
@@ -142,88 +221,82 @@
 #
 
 # We have to re-write the algorithm slightly.
-packedB = tvm.compute((N / bn, N, bn), lambda x, y, z: B[y, x * bn + z], name = 'packedB')
+packedB = tvm.compute((N1 / bn, N2, bn), lambda x, y, z: B[y, x * bn + z], name='packedB')
```
Comment (yidawang) [inline tutorials/optimize/opt_gemm.py]: Here should be `N / bn, K, bn`

**Comment by tqchen** [comment]
wait @yidawang 's approval

**Pull request merged**